File: backend/src/auth/roles.py
from __future__ import annotations

ROLE_LEVELS = {"pending": -1, "user": 0, "customer": 1, "supervisor": 2, "admin": 3}


def role_satisfies(user_role: str, required_role: str) -> bool:
    return ROLE_LEVELS.get(user_role, -1) >= ROLE_LEVELS.get(required_role, 0)

File: frontend/src/roles.js
const ROLE_LEVELS = { user: 0, customer: 1, supervisor: 2, admin: 3 }

export function roleSatisfies(userRole, requiredRole) {
  return (ROLE_LEVELS[userRole] ?? -1) >= (ROLE_LEVELS[requiredRole] ?? 0)
}
